### PR TITLE
LTS Version Change

### PR DESCRIPTION
**Pull request by Skptak**
Updating versions for LTS 2.0, added relevant PRs to the CHANGELOG

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by jasonpcarroll — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -50,18 +50,18 @@ git submodule update --checkout --init --recursive test/unit-test/CMock
 
 ## Migration Guide
 
-### How to migrate from v2.0.0 (Release Candidate) to v3.3.0
+### How to migrate from v2.0.0 (Release Candidate) to v3.4.0
```
Comment (AniruddhaKanhere) [inline README.md:53]: Is v2.0.0 still release candidate?

**Review by AniruddhaKanhere — COMMENTED** [review]
Just one nit-pick

File: README.md (hunk 1 above)
Comment (Skptak) [reply README.md:53]: I'm honestly not sure? I feel like it shouldn't be the RC given how far off that tag is. Do we just want to remove that part of it?

Comment (AniruddhaKanhere) [reply README.md:53]: I think so. I do not think that it is a release candidate now. @gitwallit can you confirm?

Comment (gitwallit) [reply README.md:53]: I believe v2.0.0 was the original release candidate version of the OTA library. Post that, we then did a GA release with v3.0.0. So v2.0.0 is still, and will aways be, the original RC version.

**Review by AniruddhaKanhere — APPROVED** [review]

**Pull request merged**